simulation/agent.py

```diff
@@ -271,9 +324,22 @@ def on_vision_update(self) -> None:
         pass
 
     @abstractmethod
-    def on_noise(self, noise: 'world.NoiseEvent') -> None:
-        """ Noise handler, will be called before `on_tick` """
-        pass
+    def on_noise(self, noises: List['world.NoiseEvent']) -> None:
+        """ Noise handler, that checks, if there a noise event occurs and where it is perceived  """
+        if self._is_deaf:
+            return
+        
+        # noises
+        for noise in noises:
+            distance = math.sqrt((noise.location.x-self.location.x)**2+(noise.location.y-self.location.y)**2)
```

Comment:
Same as above with regards to vector operations (They should all be instances of `Position` or `Vector2` IIRC)